src/shared/components/comment/comment-form.tsx

```diff
@@ -83,32 +89,35 @@ export class CommentForm extends Component<CommentFormProps, any> {
         : capitalizeFirstLetter(I18NextService.i18n.t("reply"));
   }
 
-  handleCommentSubmit(content: string, language_id?: number) {
+  async handleCommentSubmit(
+    content: string,
+    language_id?: number,
+  ): Promise<boolean> {
     const { node, onUpsertComment, edit } = this.props;
     if (typeof node === "number") {
       const post_id = node;
-      onUpsertComment({
+      return onUpsertComment({
         content,
         post_id,
         language_id,
-      });
+      }).then(r => r.state !== "failed");
     } else {
       if (edit) {
         const comment_id = node.comment_view.comment.id;
-        onUpsertComment({
+        return onUpsertComment({
           content,
           comment_id,
           language_id,
-        });
+        }).then(r => r.state !== "failed");
       } else {
         const post_id = node.comment_view.post.id;
         const parent_id = node.comment_view.comment.id;
-        this.props.onUpsertComment({
+        return onUpsertComment({
           content,
           parent_id,
           post_id,
           language_id,
-        });
+        }).then(r => r.state !== "failed");
       }
     }
```

Comment:
I think this would be cleaner as

```typescript
  async handleCommentSubmit(
    content: string,
    language_id?: number,
  ): Promise<boolean> {
    const { node, onUpsertComment, edit } = this.props;
    let response: RequestState<CommentResponse>;

    if (typeof node === "number") {
      const post_id = node;
      response = await onUpsertComment({
        content,
        post_id,
        language_id,
      });
    } else if (edit) {
        const comment_id = node.comment_view.comment.id;
        response = await onUpsertComment({
          content,
          comment_id,
          language_id,
        });
      } else {
        const post_id = node.comment_view.post.id;
        const parent_id = node.comment_view.comment.id;
        response = await onUpsertComment({
          content,
          parent_id,
          post_id,
          language_id,
        });
      }

   return response.state !== "failed";
  }
```